frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -261,6 +268,46 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	/// Greedily reduce the size of the solution to fit into the block w.r.t length.
+	///
+	/// The length of the solution is largely a function of the number of voters. The number of
+	/// winners cannot be changed. Thus, to reduce the solution size, we need to strip voters.
+	///
+	/// Note that this solution is already computed, and winners are elected based on the merit of
+	/// the total stake in the system. Nevertheless, some of the voters may be removed here.
```

Comment:
Ideal would be for this function to return `Err` if the length of the winner (`unique_targets`) changed, but that would be an overkill as it is part of the validations that we do down the road. Just document it a bit better, and good as it is.

Reply:
thanks for the thumbs up :D but: 
```suggestion
	/// the total stake in the system. Nevertheless, some of the voters may be removed here.
	/// Sometimes, removing a voter can cause a validator to also be implicitely remoted, if 
	/// that voter was the only backer of that winner. In such cases, this solution is invalid, which
	/// will be caught prior to submission. 
```

Reply:
Heh, sorry about that. I did intend to expand on the comment, but just didn't get to it yesterday when focusing on the other PR.